jbpm-services/jbpm-executor/src/test/java/org/jbpm/executor/ExtendedExecutorTest.java

```diff
@@ -0,0 +1,152 @@
+
+    private ExecutorService executorService;
+
+    private static Properties dsProps;
+
+    private EntityManagerFactory emf = null;
+
+    private PoolingDataSourceWrapper pds;
+
+    @BeforeClass
+    public static void createDBServer() {
+        System.setProperty("org.kie.executor.id", "owner");
+        System.setProperty("org.kie.executor.setDefaultOwner", "true");
+        System.setProperty("org.kie.executor.olderThan", "2");
+        dsProps = ExecutorTestUtil.getDatasourceProperties();
+        dsProps.setProperty("url", "jdbc:h2:tcp://localhost:9123/target/./jbpm-exec-test;MVCC=TRUE");
+        dsProps.setProperty("tcpPort", "9123");
+        PersistenceUtil.startH2TcpServer(dsProps);
+    }
+
+    @AfterClass
+    public static void stopDBServer() {
+        System.clearProperty("org.kie.executor.id");
+        System.clearProperty("org.kie.executor.setDefaultOwner");
+        System.clearProperty("org.kie.executor.olderThan");
+        PersistenceUtil.stopH2TcpServer();
+    }
+
+    @Before
+    public void setUp() {
+
+        pds = PersistenceUtil.setupPoolingDataSource(dsProps, "jdbc/jbpm-ds");
+        emf = Persistence.createEntityManagerFactory("org.jbpm.executor");
+
+        executorService = ExecutorServiceFactory.newExecutorService(emf);
+
+        executorService.setThreadPoolSize(1);
+        executorService.setInterval(3);
+
+        executorService.init();
+    }
+
+    @After
+    public void tearDown() {
+        executorService.clearAllRequests();
+        executorService.clearAllErrors();
+        System.clearProperty("org.kie.executor.msg.length");
+        System.clearProperty("org.kie.executor.stacktrace.length");
+        executorService.destroy();
+        if (emf != null) {
+            emf.close();
+        }
+        pds.close();
+    }
+
+    protected CountDownAsyncJobListener configureListener(int threads) {
+        CountDownAsyncJobListener countDownListener = new CountDownAsyncJobListener(threads);
+        ((ExecutorServiceImpl) executorService).addAsyncJobListener(countDownListener);
+
+        return countDownListener;
+    }
+
+    @Test(timeout = 60000)
+    public void testSchedulerWithOwner() throws Exception {
+
+        CommandContext ctxCMD = new CommandContext();
+        ctxCMD.setData("businessKey", UUID.randomUUID().toString());
+        Date schedule = new Date(Instant.now().plus(Duration.ofSeconds(5L)).toEpochMilli());
+        logger.info("schedule command for {}", schedule);
+        executorService.scheduleRequest("org.jbpm.executor.commands.ReoccurringPrintOutCommand", schedule, ctxCMD);
+        List<RequestInfo> rescheduled = executorService.getRequestsByBusinessKey((String) ctxCMD.getData("businessKey"), Arrays.asList(STATUS.QUEUED), new QueryContext());
+        assertEquals("owner", ((org.jbpm.executor.entities.RequestInfo) rescheduled.get(0)).getOwner());
+        // wait for the first two jobs to successfully run
+
+    }
+
```

Comment:
You could also add a test without executor.id (default-executor) and other with setDefaultOwner to false (owner null)

Reply:
added in another test